src/types/Posts.ts

```diff
@@ -0,0 +1,11 @@
+import { User } from './Users';
+import { Comments } from './Comments';
+
+export interface Posts {
+  userId: number;
+  id: number;
+  title: string;
+  body: string;
+  user: User | null;
+  comments: Comments[] | [];
```

Comment:
If you will be have an empty array - it not will be error
```suggestion
  comments: Comments[];
```